shared/src/main/java/net/blay09/mods/waystones/block/entity/WarpPlateBlockEntity.java

```diff
@@ -350,31 +368,70 @@ && getItem(3).getItem() == ModItems.warpDust
                 && getItem(4).getItem() == ModItems.warpDust;
     }
 
+    private boolean isReadyForTransientAttunement() {
+        return readyForAttunement
+                && getItem(0).getItem() == Items.ENDER_PEARL
+                && getItem(1).getItem() == Items.AMETHYST_SHARD
+                && getItem(2).getItem() == Items.AMETHYST_SHARD
+                && getItem(3).getItem() == Items.AMETHYST_SHARD
+                && getItem(4).getItem() == Items.AMETHYST_SHARD;
+    }
+
+
     @Nullable
-    public IWaystone getTargetWaystone() {
+    public Pair<IAttunementItem, ItemStack> getTargetAttunement() {
```

Comment:
Since `IAttunementItem` is always `ItemStack.getItem()`, it would be better to just return the ItemStack only to avoid the potential for inconsistencies.